przelewanka.ml

```diff
@@ -0,0 +1,94 @@
+(* Autor: Antoni Żewierżejew *)
+(* Reviewer: Jakub Nowak     *)
+
+(* zwraca NWD liczb a i b, gdzie nwd a 0 = a, dla a, b >= 0 *)
+let rec nwd a b =
+    if a < b then nwd b a else
+    if b = 0 then a else
+    nwd b (a mod b)
+
+open Array
+
+(* wyjątek używany do szybkiego kończenia pętli, gdy wynik zostanie znaleziony *)
+exception Znalezione of int
+
+(*
+    Jest dane n szklanek, ponumerowanych od 0 do n - 1, o pojemnościach odpowiednio x1, x2, ..., xn.
+    Początkowo wszystkie szklanki są puste. Można wykonywać następujące czynności:
+    - nalać do wybranej szklanki do pełna wody z kranu, 
+    - wylać całą wodę z wybranej szklanki do zlewu, 
+    - przelać wodę z jednej szklanki do drugiej — jeżeli się zmieści, to przelewa się całą wodę,
+      a jeżeli nie, to tyle żeby druga szklanka była pełna.
+
+    Procedura przelewanka : (int * int) array → int, mając daną tablicę par liczb
+    [|(x1, y1); (x2, y2); ...; (xn, yn)|] wyznaczy minimalną liczbę czynności potrzebnych
+    do uzyskania ze stanu (0, 0, ..., 0) stanu (y1, y2, ..., yn).
+    Jeżeli uzyskanie go nie jest możliwe, to poprawnym wynikiem jest -1. 
+
+    Założenia 0 <= n, oraz 0 <= yi <= xi dla i = 0, 1, ..., n - 1.
```

Comment:
Yoda style. :stuck_out_tongue: Lepiej napisać `n >= 0`. A właściwie to w ogóle nie pisać, niby jak miałbym przekazać do funkcji tablicę o ujemnej długości?